src/errors.ts

```diff
@@ -1,67 +1,266 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+import { WalletError as SolanaWalletError } from '@solana/wallet-adapter-base';
 import { KycDaoApiError } from './api-base';
 import { SentryConfiguration } from './types';
-import { getRandomAlphanumericString, waitForDomElement } from './utils';
+import { getRandomAlphanumericString, isLike, waitForDomElement } from './utils';
+
+function ensureType<T>() {
+  return <Actual extends T>(a: Actual) => a;
+}
 
 /**
- * Collection of error codes returned by the SDK.
- *
- * @readonly
+ * Error codes for the {@link StatusError} class
  * @enum {string}
  */
-export const KycDaoSDKErrorCodes = {
-  KycDaoApiError: 'KycDaoApiError',
-  UnexpectedError: 'UnexpectedError',
-} as const;
+export const StatusErrors = ensureType<Record<string, string>>()({
+  /** The called API function requires an active logged in user */
+  UserNotLoggedIn: 'User is not logged in',
+  /** The provided email address is invalid */
+  InvalidEmailAddress: 'Invalid email address',
+  /** The provided tax resideny was invalid */
+  InvalidTaxResidency: 'Invalid taxResidency. Please use the country list provided by the SDK.',
+  /** The called API function reqires the terms and condition to be accepted by the user */
+  TermsAndConditionsNotAccepted:
+    'Terms and Conditions and Privacy Policy must be accepted to start verification.',
+  /** The called API function reqires the disclaimer to be accepted by the user */
+  DisclaimerNotAccepted: 'Disclaimer must be accepted.',
+  /** The called API function requires a wallet to be connected */
+  WalletNotConnected: 'Wallet connection required.',
+  /** The called API function requires the user to be verified */
+  UserNotVerified: 'User must be verified to be able to mint an NFT.',
+  /** The user has an active verification while requesting a new verification */
+  UserAlreadyVerified: 'User already verified',
+  /** The selected network is not supported by the SDK */
+  NetworkNotSupported: 'Selected network is not supported',
+  /** The selected network is supported by the SDK but not enabled by the site */
+  NetworkNotEnabled: 'Selected network is not enabled',
+  /** Automatic network switching failed */
+  NetworkSwitchingFailed: 'Network switching failed',
+});
 
 /**
- * Union type of string values of {@link KycDaoSDKErrorCodes}.
- *
- * @typedef {KycDaoSDKErrorCode}
+ * Error codes for the {@link WalletError} class
+ * @enum {string}
  */
-export type KycDaoSDKErrorCode = keyof typeof KycDaoSDKErrorCodes;
+export const WalletErrors = ensureType<Record<string, string>>()({
+  /** The user is not connected to chain. It can happen because of network connecr*/
```

Comment:
This looks unfinished and I can't suggest anything because I'm not sure when we throw this :D